### PR TITLE
rgw: use 64-bit offsets for compression

### PR DESCRIPTION
**Pull request by aclamk**
This is followup of #15605
More places where improper sizes of integers were used are fixed.

### CODE REVIEW

**Comment by cbodley** [comment]
@aclamk please rebase

**Comment by cbodley** [comment]
jenkins test this please

**Comment by cbodley** [comment]
passed rgw suite: http://pulpito.ceph.com/cbodley-2017-06-19_13:44:54-rgw-wip-cbodley-testing-distro-basic-mira/

**Comment by cbodley** [comment]
seeing test failures in `unittest_rgw_compression` for `fixup_range`

File: src/rgw/rgw_compression.cc [hunk 1]
```diff
@@ -130,8 +130,8 @@ int RGWGetObj_Decompress::fixup_range(off_t& ofs, off_t& end)
     if (cs_info->blocks.size() > 1) {
       vector<compression_block>::iterator fb, lb;
       // not bad to use auto for lambda, I think
-      auto cmp_u = [] (off_t ofs, const compression_block& e) { return (unsigned)ofs < e.old_ofs; };
-      auto cmp_l = [] (const compression_block& e, off_t ofs) { return e.old_ofs <= (unsigned)ofs; };
+      auto cmp_u = [] (off_t ofs, const compression_block& e) { return (uint64_t)ofs < e.old_ofs; };
+      auto cmp_l = [] (const compression_block& e, off_t ofs) { return e.old_ofs < (uint64_t)ofs; };
```
Comment (fangyuxiangGL) [inline src/rgw/rgw_compression.cc]: should be:
`auto cmp_l = [] (const compression_block& e, off_t ofs) { return e.old_ofs <= (uint64_t)ofs; };`

Comment (aclamk) [inline src/rgw/rgw_compression.cc]: @fangyuxiangGL  ! thanks !

**Pull request merged**